docs/DeveloperGuide.adoc

```diff
@@ -153,7 +153,175 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+This will add questions based on the `Index` specified and will be added to the `slideshowQuestions` list under the `SavedQuestions` class.
+The usage of `Index` here is such that it follows the same convention of when the user edits or deletes a question.
+[NOTE]
+The order of the questions displayed on the slideshow will be *based on the input order*.
+
+*Step 2.* The command is executed and the `MainWindow` calls `CommandResult#isShowSlideshow()` to verify if the command specified
+is to start a slideshow. The `SlideshowWindow` is then displayed through the `SlideshowWindow#slideShowWindow.show()`.
+[NOTE]
+The slideshow window has already been instantiated on application launch and the window is merely being hidden or shown.
+
+*Step 3.* The window is now visible and existing questions are cleared. The list of slideshow questions is then fetched through
+`Logic#getSlideshowQuestions()` which in turn calls the `ModelManager#getSlideshowQuestions()` that fetches
+the `slideshowQuestions` list in `SavedQuestions`.
+
+*Step 4.* The user will then navigate and control the slideshow using the `Left/Right`, `Space` and `Esc` key as defined by the key
+listeners in `SlideshowWindow#initialiseKeyboardControls(Stage root)`.
+
+*Step 5.* The user exits the slideshow when the `Esc` key event is triggered or when the `currQuestionIndex` exceeds the `questionPanels.size()` or
+it falls below `0`. The behaviour of this follows the common procedure that most presentation programs adopt thus, it will not
+feel foreign to users.
+
+image::SlideshowFeatureActivityDiagram.png[]
+_Activity diagram of the actions performed_
+
+==== Design Considerations
+
+===== Controls
+
+* ** Current Implementation: **
+** The choice of using the arrow keys for navigation and the `Escape` key to quit the slideshow is such that it will feel
+familiar to users who uses presentation programs often as they have similar controls. The only difference will be the usage
+of the `Space` key to show answers as it will be something new to the users and is unique to Njoy.
+
+* ** Alternatives Considered: **
+** Usage of the `A` key to show answers. However, this is not very feasible as it is easily forgotten and not as user-friendly
+due to the smaller surface of the key as compared to the `Space` key.
+
+===== Display
+
+* ** Current Implementation: **
+** The ordering of the questions is defined based on the user input so it gives flexibility to the user to choose the ordering
+that they want the questions to be displayed.
+** Placement of the question numbering, topic, options and answer follows the common convention where
+the question numbering will be at the top followed by the topic, options and then the answer. This is such that viewers
+will not be confused by the layout.
+** *Font sizes* are displayed in the following descending order to allow the text for easy viewing:
+*** [.big]##Question Number##
+*** Question Topic & Options
+*** [.small]#Answer#
+
```

Comment:
How are the questions/answers represented in the UI structure/scene graph? Is there a particular UiPart handling each question? How are they created and organized?